pkg/handler/solana.go

```diff
@@ -0,0 +1,244 @@
+package handler
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gammazero/workerpool"
+
+	"github.com/consolelabs/indexer-api/pkg/logger"
+	"github.com/consolelabs/indexer-api/pkg/model"
+)
+
+var (
+	SolScanBaseUrl = "https://pro-api.solscan.io/v1.0"
+	SolScanToken   = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjcmVhdGVkQXQiOjE2NzU3NzcyODYxMjgsImVtYWlsIjoibmdvdHJvbmdraG9pMTEyQGdtYWlsLmNvbSIsImFjdGlvbiI6InRva2VuLWFwaSIsImlhdCI6MTY3NTc3NzI4Nn0.DCT8Fh8j9uWVpnQSMnq0uuzqeBngNLxc4r8a1Aa2C4Q"
+)
+
+func (h *Handler) MapSolanaData() error {
+	fileContent, err := os.Open("solana_collection_top_10k.json")
+	if err != nil {
+		logger.L.Fatalf(err, "Can't open file")
+		return err
+	}
+
+	defer fileContent.Close()
+
+	byteResult, _ := ioutil.ReadAll(fileContent)
+	var payload *model.SolscanCollection
+	err = json.Unmarshal(byteResult, &payload)
+	if err != nil {
+		return nil
+	}
+	wp := workerpool.New(5)
+	for _, ev := range payload.Data {
+		ev := ev
+		wp.Submit(func() {
+			h.logger.Fields(logger.Fields{
+				"collectionId": ev.CollectionID,
+			}).Infof("get data")
```

Comment:
h.logger.Fields(logger.Fields{
				"collectionId": ev.CollectionID,
			}).Infof("Enrich nft owner for collection")